mne_bids_pipeline/steps/sensor/_99_group_average.py

```diff
@@ -762,98 +770,105 @@ def average_csp_decoding(
     )
     with pd.ExcelWriter(out_files["freq"]) as w:
         grand_average_freq.to_excel(w, sheet_name="CSP Frequency", index=False)
-        grand_average_time_freq.to_excel(
-            w, sheet_name="CSP Time-Frequency", index=False
-        )
+        if grand_average_time_freq is not None:
+            grand_average_time_freq.to_excel(
+                w, sheet_name="CSP Time-Frequency", index=False
+            )
+    del grand_average_time_freq
 
     # Perform a cluster-based permutation test.
     subjects = cfg.subjects
     time_bins = np.array(cfg.decoding_csp_times)
     if time_bins.ndim == 1:
-        time_bins = np.array(list(zip(time_bins[:-1], time_bins[1:])))
-    time_bins = pd.DataFrame(time_bins, columns=["t_min", "t_max"])
-    freq_name_to_bins_map = _handle_csp_args(
-        cfg.decoding_csp_times,
-        cfg.decoding_csp_freqs,
-        cfg.decoding_metric,
-        epochs_tmin=cfg.epochs_tmin,
-        epochs_tmax=cfg.epochs_tmax,
-        time_frequency_freq_min=cfg.time_frequency_freq_min,
-        time_frequency_freq_max=cfg.time_frequency_freq_max,
-    )
-    data_for_clustering = {}
-    for freq_range_name in freq_name_to_bins_map:
-        a = np.empty(
-            shape=(
-                len(subjects),
-                len(time_bins),
-                len(freq_name_to_bins_map[freq_range_name]),
+        time_bins = np.c_[time_bins[:-1], time_bins[1:]]
+    assert time_bins.ndim == 2 and time_bins.shape[1] == 2, time_bins.shape
+    if not len(time_bins):
+        fname_csp_cluster_results = None
+    else:
+        time_bins = pd.DataFrame(time_bins, columns=["t_min", "t_max"])
```

Comment:
@hoechenberger I suggest reviewing this PR with "hide whitespace changes" because the following lines are all just indented